File: src/main/java/com/aliiensmp/aliienResize/Economy/currencyOptions/PlayerPoint.java
package com.aliiensmp.aliienResize.Economy.currencyOptions;

import com.aliiensmp.aliienResize.Economy.CurrencyProvider;
import org.black_ixx.playerpoints.PlayerPoints;
import org.black_ixx.playerpoints.PlayerPointsAPI;
import org.bukkit.entity.Player;

public class PlayerPoint implements CurrencyProvider {

    private final PlayerPointsAPI playerPointsAPI;

    /**
     * Creates the PlayerPoints adapter.
     */
    public PlayerPoint() {
        PlayerPoints instance = PlayerPoints.getInstance();
        this.playerPointsAPI = instance != null ? instance.getAPI() : null;
    }

    /**
     * Rounds a configured price up to the minimum whole number of points required.
     *
     * @param amount configured price
     * @return required point amount
     */
    private int getRequiredPoints(double amount) {
        return Math.max(0, (int) Math.ceil(amount));
    }

    @Override
    public boolean isValid() {
        return playerPointsAPI != null;
    }

    @Override
    public boolean hasBalance(Player player, double amount) {
        if (!isValid()) return false;
        return playerPointsAPI.look(player.getUniqueId()) >= getRequiredPoints(amount);
    }

    @Override
    public boolean withdraw(Player player, double amount) {
        if (!hasBalance(player, amount)) return false;
        return playerPointsAPI.take(player.getUniqueId(), getRequiredPoints(amount));
    }
}

File: src/main/kotlin/com/aliiensmp/aliienResize/economy/currencyOptions/PlayerPoint.kt
package com.aliiensmp.aliienResize.economy.currencyOptions

import com.aliiensmp.aliienResize.economy.CurrencyProvider
import org.black_ixx.playerpoints.PlayerPoints
import org.black_ixx.playerpoints.PlayerPointsAPI
import org.bukkit.entity.Player
import kotlin.math.ceil

class PlayerPoint(override val suffix: String) : CurrencyProvider {

    private val playerPointsAPI: PlayerPointsAPI?
        get() = PlayerPoints.getInstance()?.api

    override val isValid: Boolean
        get() = playerPointsAPI != null

    private fun getRequiredPoints(amount: Double): Int {
        return maxOf(0, ceil(amount).toInt())
    }

    override fun hasBalance(player: Player, amount: Double): Boolean {
        val api = playerPointsAPI ?: return false
        return api.look(player.uniqueId) >= getRequiredPoints(amount)
    }

    override fun withdraw(player: Player, amount: Double): Boolean {
        if (!hasBalance(player, amount)) return false
        return playerPointsAPI?.take(player.uniqueId, getRequiredPoints(amount)) ?: false
    }
}